src/rule_definition_tools.jl

```diff
@@ -1,4 +1,13 @@
 # These are some macros (and supporting functions) to make it easier to define rules.
+using Base.Meta
+
+macro strip_linenos(expr)
+    strip_linenos(x::LineNumberNode) = nothing
+    strip_linenos(x::Expr) = Expr(x.head, map(strip_linenos, x.args)...)
+    strip_linenos(x) = x
+    return esc(strip_linenos(expr))
```

Comment:
is this `esc` required?
I think I asked before but the answer got eaten in a force push.
(this is why ColPrac recommends against history rewriting while review is on going. Github sometimes chokes on it)

Reply:
It's still there if you click "show resolved" above.

Reply:
ah so it is is, my bad.